libvast/test/parseable.cpp

```diff
@@ -595,6 +598,92 @@ TEST(real) {
   //  CHECK(f == str.begin() + 4);
 }
 
+TEST(real - scientific) {
+  auto p = make_parser<real>{};
+  {
+    MESSAGE("null exponent");
+    auto str = ".456789e0"s;
+    real d = 0;
+    auto f = str.begin();
+    auto l = str.end();
+    CHECK(p(f, l, d));
+    CHECK_EQUAL(d, 0.456789);
+    CHECK_EQUAL(f, l);
+  }
+  {
+    MESSAGE("positive exponent");
+    auto str = ".456789e43"s;
+    real d = 0;
+    auto f = str.begin();
+    auto l = str.end();
+    CHECK(p(f, l, d));
+    CHECK_EQUAL(d, 4.56789e42);
+    CHECK_EQUAL(f, l);
+  }
+  {
+    MESSAGE("explicit positive exponent");
+    auto str = ".456789e+43"s;
+    real d = 0;
+    auto f = str.begin();
+    auto l = str.end();
+    CHECK(p(f, l, d));
+    CHECK_EQUAL(d, 4.56789e42);
+    CHECK_EQUAL(f, l);
+  }
+  {
+    MESSAGE("negative exponent");
+    auto str = ".456789e-322"s;
+    real d = 0;
+    auto f = str.begin();
+    auto l = str.end();
+    CHECK(p(f, l, d));
+    CHECK_EQUAL(d, 4.56789e-323);
+    CHECK_EQUAL(f, l);
+  }
+}
+
+// TEST(real - scientific exhaustive) {
+//   {
+//     auto p = make_parser<real>{};
+//     union real_cast {
+//       real value;
+//       struct comp {
+//         uint64_t mantissa : 52;
+//         uint16_t exponent : 11;
+//         int sign : 1;
+//       } components;
+//     };
+//     uint64_t n = 0;
+//     for (uint64_t mantissa = 0; mantissa < 10000; ++mantissa) {
+//       for (uint16_t exp = 0x400; exp < 0x7fe; ++exp, ++n) {
+//         auto cast = real_cast{
+//           .components
+//           = real_cast::comp{.mantissa = mantissa, .exponent = exp, .sign =
+//           0}};
+//         auto value = cast.value;
+//         // We wanted to trust {fmt}, but it sometimes renders wrong...
+//         auto rendered = fmt::format("{:e}", value);
+//         auto f = rendered.begin();
+//         auto l = rendered.end();
+//         real d = 0;
+//         if (!p(f, l, d))
+//           FAIL("failed to parse " << rendered);
+//         if (value != d) {
+//           const auto& ic = cast.components;
+//           const auto& oc = real_cast{.value = d}.components;
+//           FAIL(fmt::format("[{}] parser output mismatch: {} ({}) != {}\n   "
+//                            "input = {{ .mantissa = {:#015x}, .exponent = "
+//                            "{:#05x}, sign = {} }}\n  output = {{ .mantissa "
+//                            "= {:#015x}, .exponent = {:#05x}, sign = {} }}",
+//                            n, value, rendered, d, ic.mantissa, ic.exponent,
+//                            ic.sign, oc.mantissa, oc.exponent, oc.sign));
+//         }
+//       }
+//     }
+//     MESSAGE("successfully checked " << n << " generated real values");
+//   }
+// }
+
```

Comment:
Why is this commented?

Reply:
It turned out neither the `operator<<()` from libstdc++ nor `fmt::format("{:e}", value)` produce correct output for every value. I plan to submit bug reports upstream. Once these issues are fixed we can enable the test.

Reply:
Thanks. That would be nice to have as a comment in the code.